packages/temporal-time-utils/EntityWorkflow.ts

```diff
@@ -0,0 +1,74 @@
+import * as wf from "@temporalio/workflow";
+import { SignalDefinition, QueryDefinition } from "@temporalio/common";
+
+const noop = async () => {};
+type ThingToInvoke =
+  | { activity: string; activityOptions: wf.ActivityOptions }
+  | { workflow: string; workflowOptions: wf.ChildWorkflowOptions };
+export class Entity<Input = any, Update = any> {
+  MAX_ITERATIONS: number;
+  setup: (input: Input) => Promise<void>;
+  cleanup: () => Promise<void>;
+  thingToInvoke: ThingToInvoke;
+  Signal: SignalDefinition<[Update]>;
+  Query: QueryDefinition<any>;
+
+  constructor(
+    thingToInvoke: ThingToInvoke,
+    maxIterations = 1000,
+    setup = noop,
+    cleanup = noop
+  ) {
+    this.thingToInvoke = thingToInvoke;
+    this.MAX_ITERATIONS = maxIterations; // can override if needed
+    this.setup = setup;
+    this.cleanup = cleanup;
+    this.Signal = wf.defineSignal<[Update]>("EntitySignal"); // no real way to pass the types
+    this.Query = wf.defineQuery<[Update]>("EntityQuery"); // no real way to pass the types
+    this.workflow = this.workflow.bind(this);
+  }
+
+  async workflow(input: Input, isContinued = false) {
```

Comment:
How's the worker going to know the name of this workflow?
You need to export a workflow function per entity.

Reply:
this is the biggest question. perhaps i can attach a name on that workflow